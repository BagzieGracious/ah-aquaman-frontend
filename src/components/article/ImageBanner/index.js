import React, { Component } from "react";
import Banner from "../../../assets/img/banner.jpg";

export default class ImageBanner extends Component {
  render() {
    return (
      <React.Fragment>
        <section className="image-banner">
          <img src={Banner} alt="" />
          <div className="container">
            <div className="banner-content text-center">
              <h2>{this.props.info}</h2>
              <p>
                Home 
                {" "}
                <i className="fa fa-angle-right" /> 
                {" "}
                {this.props.page}
              </p>
            </div>
          </div>
        </section>
      </React.Fragment>
    );
  }
}
