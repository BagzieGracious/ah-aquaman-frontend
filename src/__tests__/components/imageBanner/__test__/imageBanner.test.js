import React from "react";
import { shallow } from "enzyme";
import ImageBanner from "../../../../components/article/ImageBanner";

describe("ArticleCard component test", () => {
  const wrapper = shallow(<ImageBanner />);

  it("Should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
