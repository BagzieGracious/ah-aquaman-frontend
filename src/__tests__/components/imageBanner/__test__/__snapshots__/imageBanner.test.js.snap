// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard component test Should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageBanner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <section
        className="image-banner"
      >
        <img
          alt=""
          src="banner.jpg"
        />
        <div
          className="container"
        >
          <div
            className="banner-content text-center"
          >
            <h2 />
            <p>
              Home
              <i
                className="fa fa-angle-right"
              />
               
            </p>
          </div>
        </div>
      </section>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src="banner.jpg"
          />,
          <div
            className="container"
          >
            <div
              className="banner-content text-center"
            >
              <h2 />
              <p>
                Home
                <i
                  className="fa fa-angle-right"
                />
                 
              </p>
            </div>
          </div>,
        ],
        "className": "image-banner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "banner.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="banner-content text-center"
            >
              <h2 />
              <p>
                Home
                <i
                  className="fa fa-angle-right"
                />
                 
              </p>
            </div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2 />,
                <p>
                  Home
                  <i
                    className="fa fa-angle-right"
                  />
                   
                </p>,
              ],
              "className": "banner-content text-center",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Home",
                    <i
                      className="fa fa-angle-right"
                    />,
                    " ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Home",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-angle-right",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " ",
                  undefined,
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <section
          className="image-banner"
        >
          <img
            alt=""
            src="banner.jpg"
          />
          <div
            className="container"
          >
            <div
              className="banner-content text-center"
            >
              <h2 />
              <p>
                Home
                <i
                  className="fa fa-angle-right"
                />
                 
              </p>
            </div>
          </div>
        </section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt=""
              src="banner.jpg"
            />,
            <div
              className="container"
            >
              <div
                className="banner-content text-center"
              >
                <h2 />
                <p>
                  Home
                  <i
                    className="fa fa-angle-right"
                  />
                   
                </p>
              </div>
            </div>,
          ],
          "className": "image-banner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "banner.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="banner-content text-center"
              >
                <h2 />
                <p>
                  Home
                  <i
                    className="fa fa-angle-right"
                  />
                   
                </p>
              </div>,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2 />,
                  <p>
                    Home
                    <i
                      className="fa fa-angle-right"
                    />
                     
                  </p>,
                ],
                "className": "banner-content text-center",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Home",
                      <i
                        className="fa fa-angle-right"
                      />,
                      " ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Home",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fa fa-angle-right",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " ",
                    undefined,
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "section",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
